### DISCUSSION TITLE
Corrupted mariadb database.  Any way to salvage instead of rebuild from scratch

### DISCUSSION THREAD

**Discussion by jalin2** [photoprism/photoprism#1959] [General]
Everything looks like it still runs properly but when I check the photoprism logs I see that there are messages related to refusal of connection to the database.  Upon more investigation it looks like a table in the database is corrupt.  The markers table by the looks of it?

I tried to backup the database and it's not backing up properly and instead sends db container to restart which sounds like it's related to db being corrupted.

I have albums, people, favorites and moments that ideally I'd like to retain without having to rebuild the whole db again.

Is there a way to do so?  Assuming if it was something in the marker table which is corrupt, can I drop table and recreate as empty and then reindex somehow or anything like that?

**Comment by lastzero** [comment]
Doesn't https://docs.photoprism.app/getting-started/troubleshooting/mariadb/#corrupted-files help?

**Reply by jalin2** [reply]
Hi @lastzero thank for getting back to me.  I did try with --innodb-force-recovery=1 setting  and was still not able to back up the database properly.  During backup it would still crash and the logs would indicate it crashed during dump from marker table at some row.

I ended up dropping the markers table and then recreating the table with no entries.  After doing so, backup worked again.  I then did a 'faces reset' and 'faces index'.  Although I have to rebuild the people again I'm happy that I got to keep the albums, favorites, ...

I've now included a photoprism backup in a cron job which will hopefully save me some trouble in the future.

Btw, thanks for building this wonderful software.
